scene/main/viewport.cpp

```diff
@@ -1403,10 +1403,6 @@ Control *Viewport::gui_find_control(const Point2 &p_global) {
 }
 
 Control *Viewport::_gui_find_control_at_pos(CanvasItem *p_node, const Point2 &p_global, const Transform2D &p_xform, Transform2D &r_inv_xform) {
-	if (Object::cast_to<Viewport>(p_node)) {
-		return nullptr;
-	}
-
```

Comment:
Actually got one concern: is `p_node` guaranteed to be not null in here? If not then previously that removed `if` would handle such case and now it would crash in the subsequent `if`.

Reply:
There are two locations in the code-base, where `_gui_find_control_at_pos` is called. Both locations check beforehand, that `p_node` is not null.

Reply:
Then all good. :slightly_smiling_face: